mpc_local_planner/src/mpc_local_planner_ros.cpp

```diff
@@ -859,6 +859,13 @@ teb_local_planner::RobotFootprintModelPtr MpcLocalPlannerROS::getRobotFootprintF
         return boost::make_shared<teb_local_planner::PointRobotFootprint>();
     }
 
+    // from costmap_2d
+    if (model_name.compare("costmap_2d") == 0)
+    {
+        ROS_INFO("Footprint model loaded from costmap_2d for trajectory optimization.");
```

Comment:
By taking my previous comment into account, we can move the check from line 989-993 for validity to this line and return a PointModel as default. This also prevents the planner from printing both messages in case of invalidity:

     "Footprint model loaded from costmap_2d for trajectory optimization."
     "Costmap 2d pointer is null. Using point model instead."

Reply:
I think I got it. Let me know if there's anything else.